### PR TITLE
Amsinh/d tr milestone1

### PR DESCRIPTION
**Pull request by ameyanator**

### CODE REVIEW

File: src/Microsoft.ServiceFabric.Services.Remoting/V2/ActivityIdLogicalCallContext.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        /// <returns><c>true</c> if this instance is present; otherwise, <c>false</c>.</returns>
+        internal bool IsPresent()
+        {
+            return Activity.Current != null;
+        }
+
+        /// <summary>
+        /// Tries the get.
+        /// </summary>
+        /// <returns><c>true</c> if XXXX, <c>false</c> otherwise.</returns>
```
Comment (PavanKunapareddyMSFT) [inline src/Microsoft.ServiceFabric.Services.Remoting/V2/ActivityIdLogicalCallContext.cs:87]: Not true

File: src/Microsoft.ServiceFabric.Services.Remoting/V2/ActivityIdLogicalCallContext.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            {
+                activity.SetParentId(parentId);
+                if (requestMessage.GetHeader().ActivityIdTraceStateHeader != null)
+                {
+                    activity.TraceStateString = requestMessage.GetHeader().ActivityIdTraceStateHeader;
+                }
+            }
+            else
+            {
+                Activity.DefaultIdFormat = ActivityIdFormat.W3C;
```
Comment (PavanKunapareddyMSFT) [inline src/Microsoft.ServiceFabric.Services.Remoting/V2/ActivityIdLogicalCallContext.cs:54]: If there is a parent, we dont start the activity?

File: src/Microsoft.ServiceFabric.Services.Remoting/V2/ActivityIdLogicalCallContext.cs [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            string parentId = null;
+            if (requestMessage.GetHeader().ActivityIdParent != null)
+            {
+                Activity.DefaultIdFormat = ActivityIdFormat.W3C;
+                Activity.ForceDefaultIdFormat = true;
+                parentId = requestMessage.GetHeader().ActivityIdParent;
+            }
+            else
+            {
+                parentId = requestMessage.GetHeader().ActivityRequestId;
```
Comment (PavanKunapareddyMSFT) [inline src/Microsoft.ServiceFabric.Services.Remoting/V2/ActivityIdLogicalCallContext.cs:41]: In both the cases, we are setting the parent. Is HttpListener code is also doing the same? If ActicityIdParent is used to identify W3C format, we can probably send the format itself?

File: src/Microsoft.ServiceFabric.Services.Remoting/V2/FabricTransport/Client/FabricTransportServiceRemotingClient.cs [hunk 4]
```diff
@@ -105,6 +113,37 @@ public Task OpenAsync(CancellationToken cancellationToken)
         public async Task<IServiceRemotingResponseMessage> RequestResponseAsync(
             IServiceRemotingRequestMessage remotingRequestRequestMessage)
         {
+#if DotNetCoreClr
```
Comment (PavanKunapareddyMSFT) [inline src/Microsoft.ServiceFabric.Services.Remoting/V2/FabricTransport/Client/FabricTransportServiceRemotingClient.cs:116]: Are we not supporting this in NetFx?

File: src/Microsoft.ServiceFabric.Services.Remoting/V2/FabricTransport/Client/FabricTransportServiceRemotingClient.cs [hunk 5]
```diff
@@ -105,6 +113,37 @@ public Task OpenAsync(CancellationToken cancellationToken)
...
+                // Activity ID is not present, make one of W3C format
+                currentActivity = this.activityIdLogicalCallContext.CreateW3CActivity("Call from Request Response Async");
+                currentActivity.Start();
+            }
+            else
+            {
+                currentActivity = this.activityIdLogicalCallContext.TryGet();
+            }
+
+            if (currentActivity.IdFormat == ActivityIdFormat.W3C)
```
Comment (PavanKunapareddyMSFT) [inline src/Microsoft.ServiceFabric.Services.Remoting/V2/FabricTransport/Client/FabricTransportServiceRemotingClient.cs:130]: NIT: Have an utility method to construct the headers.

File: src/Microsoft.ServiceFabric.Services.Remoting/V2/ActivityIdLogicalCallContext.cs [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        /// <returns>Activity.</returns>
+        internal Activity CreateW3CActivity(string activityMessage = "Create new Activity")
+        {
+            Activity.DefaultIdFormat = ActivityIdFormat.W3C;
+            Activity.ForceDefaultIdFormat = true;
+            var activity = new Activity(activityMessage);
+            return activity;
+        }
+
+        internal Activity StartActivity(IServiceRemotingRequestMessage requestMessage, string activityMessage = "Start new Activity")
```
Comment (PavanKunapareddyMSFT) [inline src/Microsoft.ServiceFabric.Services.Remoting/V2/ActivityIdLogicalCallContext.cs:29]: NIT: Could be static methods. We do not use any class properties here.

File: nuprojs/Microsoft.ServiceFabric.Services.Remoting/Microsoft.ServiceFabric.Services.Remoting.nuproj [hunk 7]
```diff
@@ -98,5 +98,8 @@
       <Dependency Include="Microsoft.ServiceFabric.FabricTransport.Internal">
         <Version>[$(NugetPkg_Version_Microsoft_ServiceFabric_FabricTransport_Internal)]</Version>
       </Dependency>
+	  <Dependency Include="System.Diagnostics.DiagnosticSource">
+		<Version>[5.0.0.0]</Version>
```
Comment (PavanKunapareddyMSFT) [inline nuprojs/Microsoft.ServiceFabric.Services.Remoting/Microsoft.ServiceFabric.Services.Remoting.nuproj:102]: Need framework dependent nuget inclusion

**Review by PavanKunapareddyMSFT — CHANGES_REQUESTED** [review]

File: src/Microsoft.ServiceFabric.Services.Remoting/V2/ActivityIdLogicalCallContext.cs (hunk 3 above)
Comment (ameyanator) [reply src/Microsoft.ServiceFabric.Services.Remoting/V2/ActivityIdLogicalCallContext.cs:41]: Http does the same

File: src/Microsoft.ServiceFabric.Services.Remoting/V2/ActivityIdLogicalCallContext.cs (hunk 2 above)
Comment (ameyanator) [reply src/Microsoft.ServiceFabric.Services.Remoting/V2/ActivityIdLogicalCallContext.cs:54]: We add the other information, and start it at the end

File: src/Microsoft.ServiceFabric.Services.Remoting/V2/FabricTransport/Client/FabricTransportServiceRemotingClient.cs (hunk 4 above)
Comment (ameyanator) [reply src/Microsoft.ServiceFabric.Services.Remoting/V2/FabricTransport/Client/FabricTransportServiceRemotingClient.cs:116]: Activity.cs is not supported by Net Framework

File: src/Microsoft.ServiceFabric.Services.Remoting/V2/ActivityIdLogicalCallContext.cs (hunk 3 above)
Comment (ameyanator) [reply src/Microsoft.ServiceFabric.Services.Remoting/V2/ActivityIdLogicalCallContext.cs:41]: Link to reference -> https://github.com/vancem/AspNetCore/blob/38ac3da95426132c7f90bfc88e02e308d151eea3/src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs#:~:text=activity.SetParentId(requestId)%3B

File: nuprojs/Microsoft.ServiceFabric.Services.Remoting/Microsoft.ServiceFabric.Services.Remoting.nuproj (hunk 7 above)
Comment (ameyanator) [reply nuprojs/Microsoft.ServiceFabric.Services.Remoting/Microsoft.ServiceFabric.Services.Remoting.nuproj:102]: Just trying to understand, does framework here mean between netcore and netframework? or between net5.0 and netcore3.1 such

File: src/Microsoft.ServiceFabric.Services.Remoting/V2/ActivityIdLogicalCallContext.cs (hunk 1 above)
Comment (ameyanator) [reply src/Microsoft.ServiceFabric.Services.Remoting/V2/ActivityIdLogicalCallContext.cs:118]: Have an IsPresent method. So changed this to Get()

**Pull request closed without merging**